disable_warnings.sh

```diff
@@ -0,0 +1,38 @@
+#floriantschopp/schneith, 07/2019
+#!/bin/bash
+
+# Check for arguments
+if [ -z "$1" ]; then
+  echo "Disable the warnings generated by the eigen core headers by adding pragma's to all header files.";
+  echo -e "Usage: disable_warnings.sh CATKIN_DEVEL_SPACE/EIGEN_SYSTEM_INCLUDE"
+  exit -1
+fi
+
+DEVEL_SPACE=$1
+EIGEN_INCLUDES_DIR=$1/include/eigen3
+CURR_DIR=$(pwd -P)
+TEMPLATE_DIR=$(pwd -P)/template
+cd $EIGEN_INCLUDES_DIR
+CORE_DIR=$(pwd -P)/Eigen/src/Core
+cd $CURR_DIR
+
+echo "Attempt to change headers in $CORE_DIR"
+
+# Now check if gtsam headers are installed and the path is correct
+if [ ! -d "$EIGEN_INCLUDES_DIR" ] || [ ! -f "$CORE_DIR/ArrayBase.h" ]; then
+  echo "Could not find eigen headers in the specified devel space!";
+  exit -1
+fi
+
+# Check wheter the warnings are already disabled
+ALREADY_IGNORED=$(cat $CORE_DIR/ArrayBase.h | grep "pragma GCC diagnostic ignored \"-Wint-in-bool-context\"" | wc -l)
+if [ "$ALREADY_IGNORED" -eq "1" ]; then
```

Comment:
Instead of counting with `wc -l` you can directly use `grep`'s exit code it indicates whether it found something or not. E.g. the following replaces the above two lines:

```sh
if grep -q "pragma GCC diagnostic ignored \"-Wint-in-bool-context\"" $CORE_DIR/ArrayBase.h; then
```

The `-q` (for quiet) is to prevent any output of matches.
Actually there is a little differences in effect: you require exactly one match while my suggestion accepts multiple matches. I assume the latter is what you want because otherwise it will only continue adding headers once there are two matches or more for some wired reason.